### ISSUE TITLE
error: invalid resource name "kubernetes-dashboard-349c57c89c-bk344\npods/kubernetes-dashboard-349c57c89c-gbrcd": [may not contain '/']

### ISSUE THREAD

**Issue by Zimmergren** [Azure/AKS#214]
I've set up 4 different AKS clusters, and with the last one I bumped into issues after it had been running for a while. I haven't seen this with any of the other clusters. 

**Command executed**
```
az aks browse -g myGroup -n myClusterName
```

**Response**
```
Merged "myClusterName" as current context in C:\Users\tobia\AppData\Local\Temp\tmp7fzdq5nj
Proxy running on http://127.0.0.1:8001/
Press CTRL+C to close the tunnel...
error: invalid resource name "kubernetes-dashboard-349c57c89c-bk344\npods/kubernetes-dashboard-349c57c89c-gbrcd": [may not contain '/']
```

**Additional background**
I've put this specific cluster to the test by increasing the number of replicas per deployment to max it out. After a few minutes the dashboard became unresponsive, then I couldn't connect anymore. 
However, after that I have used `kubectl delete -f mydefinition.yml` to remove the deployment again, and I can verify with kubectl that the deployment is no longer there - but I still can't connect to the dashboard.

**Comment by Zimmergren** [comment]
I've got a tonne of more related issues to the above, but I suppose an interesting piece of information is that this happens, and likely why I can't access the cluster anymore.

It looks like everything is becoming unhealthy, then just drops. This is a recurring issue that I've managed to reproduce across at least 5 AKS clusters now, and counting. With that, I wouldn't consider it fit for production just yet.

```
λ kubectl get cs
NAME                 STATUS      MESSAGE                                                                                        ERROR
controller-manager   Unhealthy   Get http://127.0.0.1:10252/healthz: dial tcp 127.0.0.1:10252: getsockopt: connection refused
scheduler            Unhealthy   Get http://127.0.0.1:10251/healthz: dial tcp 127.0.0.1:10251: getsockopt: connection refused
etcd-0               Healthy     {"health": "true"}
```

**Comment by Zimmergren** [comment]
@antoinne85 since nobody replied here for more than a month, I took the "easy" way out and just re-created the cluster. I have seen this issue a couple of time since, and I am not sure about the best way to handle it. I've got some slick routines for quickly setting up an exact replica of my cluster, so it wasn't a big deal - but I would still like to learn the cause and solution for this.

**Comment by mboersma** [comment]
@Zimmergren @antoinne85 the "invalid resource name" issue looks like a client-side error actually, one that should have been fixed by #5462. 

Have you tried this with a CLI 2.0.26 or newer?

Also, we've seen similar problems when `kubectl` is out of date relative to the Kubernetes cluster, so make sure they are the same version.
